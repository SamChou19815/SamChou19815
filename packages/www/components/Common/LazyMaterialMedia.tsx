import React, { ReactElement } from 'react';

import CardMedia from '@material-ui/core/CardMedia';

type Props = {
  readonly image: string;
  readonly title: string;
};

export default ({ image, title }: Props): ReactElement => (
  <CardMedia image={image} title={title} component="img" loading="lazy" />
);
