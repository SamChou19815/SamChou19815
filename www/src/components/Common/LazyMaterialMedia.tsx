import React, { ReactElement } from 'react';
import CardMedia from '@material-ui/core/CardMedia';

type Props = {
  readonly image: string;
  readonly title: string;
};

export default ({ image, title }: Props): ReactElement => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  // @ts-ignore
  <CardMedia image={image} title={title} component="img" loading="lazy" />
);
