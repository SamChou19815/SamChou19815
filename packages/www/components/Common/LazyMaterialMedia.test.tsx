import React from 'react';

import renderer from 'react-test-renderer';

import LazyMaterialMedia from './LazyMaterialMedia';

it('LazyMaterialMedia matches snapshot.', () => {
  const tree = renderer.create(<LazyMaterialMedia image="test" title="title" />).toJSON();
  expect(tree).toMatchSnapshot();
});
