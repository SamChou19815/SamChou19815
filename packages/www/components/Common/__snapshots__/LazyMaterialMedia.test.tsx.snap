// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyMaterialMedia matches snapshot. 1`] = `
<img
  className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
  loading="lazy"
  src="test"
  title="title"
/>
`;
